typed-racket-lib/typed-racket/utils/shallow-contract.rkt

```diff
@@ -46,23 +46,19 @@
           (else ;#(keyword<? actual expected)
            (loop expected-kw* (cdr actual-kw*)))))))
 
-(define (shallow-and/c . pred*)
-  (lambda (x)
-    (let loop ((p?* pred*))
-      (if (null? p?*)
-        #true
-        (if ((car p?*) x)
-          (loop (cdr p?*))
-          #false)))))
+(define ((shallow-and/c . pred*) x)
```

Comment:
These functions could both be converted to `for` loops automatically (for/and and for/or respectively).

Reply:
Filed jackfirth/resyntax#454.